internal/devices/create.go

```diff
@@ -78,48 +78,66 @@ func (c *Client) Create() *cobra.Command {
 				}
 				userdata = string(userdataRaw)
 			}
+			var endDt time.Time
 
 			if terminationTime != "" {
 				parsedTime, err := time.Parse(time.RFC3339, terminationTime)
 				if err != nil {
 					return fmt.Errorf("Could not parse time %q: %w", terminationTime, err)
 				}
-				endDt = &packngo.Timestamp{Time: parsedTime}
+				endDt = parsedTime
 			}
 
 			var facilityArgs []string
+			var request metal.CreateDeviceRequest
+			spm := float32(spotPriceMax)
+
 			if facility != "" {
 				facilityArgs = append(facilityArgs, facility)
+
+				deviceCreateInFacilityInput := &metal.DeviceCreateInFacilityInput{
+					Plan:                  plan,
+					OperatingSystem:       operatingSystem,
+					Hostname:              &hostname,
+					BillingCycle:          &billingCycle,
+					Userdata:              &userdata,
+					IpxeScriptUrl:         &ipxescripturl,
+					AlwaysPxe:             &alwaysPXE,
+					HardwareReservationId: &hardwareReservationID,
+					SpotInstance:          &spotInstance,
+					SpotPriceMax:          &spm,
+					TerminationTime:       &endDt,
+					Facility:              facilityArgs,
+				}
+				request = metal.CreateDeviceRequest{DeviceCreateInFacilityInput: deviceCreateInFacilityInput, DeviceCreateInMetroInput: nil}
```

Comment:
@ctreatma, 

As per the document, createDevice_request allows only one input. But the actual API allows both inputs(facility, Metro).
```
 createDeviceRequest := openapiclient.createDevice_request{DeviceCreateInFacilityInput: openapiclient.NewDeviceCreateInFacilityInput([]string{"Facility_example"}, "OperatingSystem_example", "c3.large.x86")} // CreateDeviceRequest | Device to create
```
Packngo is successfully runs the below and creates a device, But metal-go throws the below error.
```
vasu@vasu-cloud:~/Equinix/metal-cli/bin$ ./metal device create -p 65dcd8f4-abd4-4e7a-8b29-da4b030edd96 -P c3.medium.x86 -m sv -H test-metalGo-2 -O ubuntu_20_04
++++ METRO ++++Error: Could not create Device: 422 Unprocessable Entity

```

Reply:
With `metal-go` you have to construct a slightly different request depending on whether the facility or metro flag was used.  This is how it's done in the cluster-api provider: https://github.com/kubernetes-sigs/cluster-api-provider-packet/pull/527/files#diff-febb66857beae6d70b0aad1c73c92369afe1fb7d309f699f6047af7912d99e05R159-R183

It is possible to restructure the spec for a more `packngo`-like experience, but given that facility is going away in a few months anyway, it's not really worth doing that restructuring right now.